plugin/trino-memory/src/test/java/io/trino/plugin/memory/TestMemorySmoke.java

```diff
@@ -107,6 +113,40 @@ public void testSelect()
         assertQueryResult("SELECT count(*) FROM test_select", 75L);
     }
 
+    @Test
+    public void testCustomMetricsScanFilter()
+    {
+        Map<String, Metric> metrics = collectCustomMetrics("SELECT partkey FROM part WHERE partkey % 1000 > 0");
```

Comment:
Would it be possible to avoid using raw types here?

Reply:
Sounds good - do you mean returning a `Metrics` object from `collectCustomMetrics()` (instead of a `Map`)?
If so, I think it can be fixed by https://github.com/rzeyde-varada/trino/commit/7332dbeb4935f5c7781cfc23f4de4cba4a2975ca.

Reply:
I don't know this code, so cannot suggest something.
Just my IDE warns me that `Metric` is a generic class, but here it's used without `<...>`.

maybe `Metric<?>` would do for now

Reply:
Opened #8707.